embedded-google-pubsub/src/main/java/com/playtika/test/pubsub/EmbeddedPubsubBootstrapConfiguration.java

```diff
@@ -88,8 +90,12 @@ private void registerPubsubEnvironment(GenericContainer container,
         environment.getPropertySources().addFirst(propertySource);
     }
 
-    @Bean
-    public PubSubResourcesGenerator pubSubResourcesGenerator(GenericContainer pubsub, PubsubProperties properties) throws IOException {
-        return new PubSubResourcesGenerator(pubsub, properties, properties.getProjectId());
+    @Bean(name = PubsubProperties.BEAN_NAME_EMBEDDED_GOOGLE_PUBSUB_RESOURCES_GENERATOR)
+    public PubSubResourcesGenerator pubSubResourcesGenerator(@Qualifier(PubsubProperties.BEAN_NAME_EMBEDDED_GOOGLE_PUBSUB) GenericContainer pubsub,
+                                                             PubsubProperties properties) throws IOException {
+        PubSubResourcesGenerator pubSubResourcesGenerator = new PubSubResourcesGenerator(pubsub, properties);
+        pubSubResourcesGenerator.init();
```

Comment:
you may want spring to handle initialization of your bean, if yes, change ```@Bean``` properties